issue205 skipws propagation (regression tests)

goto40 opened the pull request:
Here, we add some tests to check and demonstrate the propagation of skipws/noswkips across rules.

We should consider to add this tests, once the Arpeggio fix for the Arpeggio issue 61 is done.

<!-- Please don't remove/change code review checklist -->

## Code review checklist

- [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [x] Title summarizes what is changing
- [n/a] Commit messages are meaningful (see [this][commit messages] for details) - squash
- [x] Tests have been included and/or updated
- [n/a] Docstrings have been included and/or updated, as appropriate
- [x] Standalone docs have been updated accordingly
- [n/a] Changelog(s) has/have been updated, as needed (see `CHANGELOG.md`).


[commit messages]: https://chris.beams.io/posts/git-commit/

igordejanovic commented:
@goto40 Arpeggio bugfix release 1.9.1 is out with a regression test for the issue. 

I don't think that we also need regression tests in textX as well as it was solely an Arpeggio issue but I suggest, since you have spent time to investigate `skipws` behavior, to move some of these tests (I would prefer a non-redundant subset) into standard battery of tests to cover `skipws` better.

goto40 commented:
OK. Great. I will rework the test cases.

goto40 committed:
Add note in the documentation about rule modifier propagation

goto40 commented:
* I reworked the tests and the documentation alongside the test cases.
 * I added some explanation to the docs (grammar.md)

Feel free to comment...

**However**: I installed the previous version of Arpeggio (```git co aff1bb8```) and all tests pass. I cannot (re)produce an error showing the new Arpeggio version has an effect... Any idea, what to do to demonstrate the old Arpeggio version was a problem for textx?

igordejanovic commented:
@goto40 Created a regression test that fails with the unfixed version of Arpeggio and pass with fixed version. I think that this single test is all what we need. I've reused your docstrings for explanation in comment. If you agree with the change I would merge/cherry pick just this single commit and your addition to the docs.

goto40 commented:
Very good! Good work.
OK for me, go ahead as you described.

igordejanovic committed:
refs #205 Add regression test for skipws/noskipws issue
